traffic_ops/testing/api/v4/serverupdatestatus_test.go

```diff
@@ -404,44 +419,77 @@ func TestSetServerUpdateStatuses(t *testing.T) {
+			}
+			if configApply != nil {
+				if afterServer.ConfigApplyTime == nil || !afterServer.ConfigApplyTime.Equal(*configApply) {
+					t.Errorf("Failed to set server's ConfigApplyTime. expected: %v actual: %v", *configApply, afterServer.ConfigApplyTime)
 				}
 			}
-			if reval != nil {
-				if *newServer[0].RevalPending != *reval {
-					t.Errorf("set reval update pending to %v, but then got server %v", *reval, *newServer[0].RevalPending)
+			if revalUpdate != nil {
+				if afterServer.RevalUpdateTime == nil || !afterServer.RevalUpdateTime.Equal(*revalUpdate) {
+					t.Errorf("Failed to set server's RevalUpdateTime. expected: %v actual: %v", *revalUpdate, afterServer.RevalUpdateTime)
 				}
-			} else {
-				if *newServer[0].RevalPending != *existingServer[0].RevalPending {
-					t.Errorf("set reval update pending with nil (don't update), but then got server %v which didn't match pre-update value %v", *newServer[0].RevalPending, *existingServer[0].RevalPending)
+			}
+			if revalApply != nil {
+				if afterServer.RevalApplyTime == nil || !afterServer.RevalApplyTime.Equal(*revalApply) {
+					t.Errorf("Failed to set server's RevalApplyTime. expected: %v actual: %v", *revalApply, afterServer.RevalApplyTime)
+				}
+			}
+
+			// Ensure boolean logic continues to work as expected
+			if configUpdate != nil && configApply != nil {
+				if ((*configUpdate).Before(*configApply) || (*configUpdate).Equal(*configApply)) &&
+					*afterServer.UpdPending {
+					t.Error("The configUpdateTime <= configApplyTime. UpdPending should be false")
+				} else if (*configUpdate).After(*configApply) && !*afterServer.UpdPending {
+					t.Error("The configUpdateTime > configApplyTime. UpdPending should be true")
+				}
+			}
+			if revalUpdate != nil && revalApply != nil {
+				if ((*revalUpdate).Before(*revalApply) || (*revalUpdate).Equal(*revalApply)) &&
+					*afterServer.RevalPending {
+					t.Error("The configUpdateTime <= configApplyTime. RevalPending should be false")
+				} else if (*revalUpdate).After(*revalApply) && !*afterServer.RevalPending {
+					t.Error("The configUpdateTime > configApplyTime. RevalPending should be true")
 				}
 			}
 		}
 
-		testVals(util.BoolPtr(true), util.BoolPtr(true))
-		testVals(util.BoolPtr(true), util.BoolPtr(false))
-		testVals(util.BoolPtr(false), util.BoolPtr(false))
-		testVals(nil, util.BoolPtr(true))
-		testVals(nil, util.BoolPtr(false))
-		testVals(util.BoolPtr(true), nil)
-		testVals(util.BoolPtr(false), nil)
+		// Postgres stores microsecond precision. There is also some discussion around MacOS losing
+		// precision as well. The nanosecond precision is accurate within go one linux however,
+		// but round trips to and from the database may result in an inaccurate Equals comparison
+		// with the loss of precision. Also, it appears to Round and not Truncate.
+		now := time.Now().Round(time.Microsecond)
+		later := now.Add(time.Hour * 6)
```

Comment:
So this is what I ran into testing:
My Mac's precision is microseconds. And Postgres' precision is also in microseconds according to the docs. When running my integration tests locally on my machine, the microsecond resolution on my Mac matched that precision to the returned time from the server+Postgres. However, when run on a Linux machine we have down to the nanosecond precision with a monotonic clock reading. The Linux test would see a greater precision (either higher or lower by nanoseconds+monotonic) than the result returned via the roundtrip server+postgres. I was seeing results like:
```
2022-03-23T21:30:36.535713737-06:00 // Linux container test values
2022-03-23T21:30:36.535714-06:00 // Response from server since it was rounded to Microseconds by Postgres
```